src/brute_tree.jl

```diff
@@ -22,8 +22,8 @@ function BruteTree(data::AbstractVector{V}, metric::Metric = Euclidean();
     BruteTree(storedata ? data : Vector{V}(), metric, reorder)
 end
 
-function BruteTree(data::AbstractMatrix{T}, metric::Metric = Euclidean();
-                   reorder::Bool=false, leafsize::Int=0, storedata::Bool=true) where {T}
+function BruteTree(data::AbstractVecOrMat{T}, metric::Metric = Euclidean();
+                   reorder::Bool=false, leafsize::Int=0, storedata::Bool=true) where {T <: AbstractFloat}
```

Comment:
Is the new `T <: AbstractFloat` relevant for this PR?

Reply:
I believe I thought it was necessary to distinguish it from `BruteTree(data::AbstractVector{V}...) where {V <: AbstractVector}`, but I see that's probably not true. Both BallTree and KDTree currently provide that same type constraint. Doesn't look like it's necessary for BallTree or KDTree either.